### PR TITLE
modules/aws: limit docker log file sizes

### PR DESCRIPTION
**Pull request by sym3tri**
Untested. Do NOT merge yet.

First step towards fixing #221

### CODE REVIEW

**Comment by sym3tri** [comment]
Handing this over to @squat

**Comment by chancez** [comment]
So the current commit isn't quite right, it's missing quotes around the value.

I used this and my nodes came up successfully:
```diff
+
+  dropin = [
+    {
+      name    = "10-dockeropts.conf"
+      content = "[Service]\nEnvironment=\"DOCKER_OPTS=--log-opt max-size=50m --log-opt max-file=3\"\n"
+    },
+  ]
```

I'll respond shortly once I've determined it actually limits the log sizes. It might be a little before I can personally say it works with fluentd, but I'm actually not all that worried about that.

**Comment by chancez** [comment]
It might be a good idea to parameterize these, and set some reasonable defaults. I do expect some customers will need to tweak these settings since it has an impact on if logs get deleted before they're collected, depending on log throughput and collection throughput.

**Comment by sym3tri** [comment]
@chancez yeah I agree. But first let's just get this in to avoid the urgent disk issues we've been seeing. We can parameterize later in a separate PR

**Comment by s-urbaniak** [comment]
@sym3tri set the `do-not-merge` label for now as per initial comment. Please ping once we can get this in. Thanks!

**Comment by sym3tri** [comment]
@squat thanks for fixing & testing. Is this ready to remove the "WIP" and "do-not-merge"?

**Comment by squat** [comment]
yes, just finished testing and it is all looking good!

**Comment by squat** [comment]
rebased and ready to merge @sym3tri

**Comment by sym3tri** [comment]
@squat great. Can you summarize how you tested/validated the changes?

**Comment by squat** [comment]
I primarily tested that the PR produces an ignition file that results in a valid configuration for Docker. To test this, I verified that the  `--log-opt max-size` parameter actually limits the logs size.

To reproduce:

* modify log options to be `--log-opt max-size=1k`
* terraform apply to create cluster
* ssh onto node
* test:
    ```shell
    # run test command to generate lots of logs:
    docker run -d --name=test -t busybox /bin/sh -c 'n=0; while true; do n=$((n+1));echo $n; done'
    sleep 2s && docker kill test
    # show all the logs. verify that this only every shows a few (~10) lines; most of the time this shows nothing as the logs are constantly being rotated:
    docker logs test --tail=all
    docker rm test
    # change log opts back to `50m`
    sudo vim /etc/systemd/system/docker.service.d/10-dockeropts.conf
    # restart docker:
    sudo systemctl daemon-reload
    sudo systemctl restart docker
    # run test command again:
    docker run -d --name=test -t busybox /bin/sh -c 'n=0; while true; do n=$((n+1));echo $n; done'
    sleep 2s && docker kill test
    # show all the logs. verify that all the data is shown:
    docker logs test --tail=all
    docker rm test
    ```

**Comment by squat** [comment]
alternatively:

* modify log options to be `--log-opt max-size=1k`
* terraform apply to create cluster
* ssh onto node
* verify all logs are smaller than 1k:
    ```sh
    ls -hal /var/lib/docker/containers/*/*.log | awk '{print $5}'
    ```

**Comment by chancez** [comment]
I've also tested my own patch of this which is identical. Fluentd/kubelet also handle the log rotation just fine.

**Comment by sym3tri** [comment]
LGTM
